### PR TITLE
Fix flickering effects in individual blog page

### PR DESCRIPTION
**Pull request by IsAmrish**
Signed-off-by: isamrish <askmaurya48@gmail.com>

This commit will
- fix flickering effects. It was due to updatation of the state of the component
- fix the padding issue for the big screen.

Padding issue:

Previous:

prev

After:

after

### CODE REVIEW

File: website/src/pages/Blog/Author/index.tsx [hunk 1]
```diff
@@ -45,17 +45,17 @@ const Blog: React.FC = () => {
...
-      author: "",
-      excerpt: "",
-      author_info: "",
-      date: "",
-      tags: [""],
-      content: "",
-      id: 0,
-      slug: "" }
-  ]);
+  const [authorArticlesData, setAuthorArticlesData] = useState<blog[] | null>([
```
Comment (Pallavi-PH) [inline website/src/pages/Blog/Author/index.tsx:48]:
```suggestion
  const [authorBlogsData, setAuthorBlogData] = useState<blog[] | null>([
```

For consistent naming

File: website/src/pages/Blog/Author/index.tsx [hunk 2]
```diff
@@ -89,56 +88,56 @@ const Blog: React.FC = () => {
...
-      const { author, author_info } = authorData!;
-      const image = `${currentOrigin}/images/blog/authors/${getAvatar(filteredAuthorData[0]?.author)}.png`;
+    if(authorArticlesData && authorArticlesData.length && currentOrigin && authorName) {
+      const { author, author_info } = authorArticlesData[0]!;
+      const image = `${currentOrigin}/images/blog/authors/${getAvatar(authorArticlesData[0]?.author)}.png`;
       const url = `${currentOrigin}/blog/author/${authorName}`;
       setAuthorMetadata({ author, author_info, image, url });
     }
-  }, [filteredAuthorData, currentOrigin, authorName]);
+  }, [authorArticlesData, currentOrigin, authorName]);
```
[the remark this answers is not in the document]
Comment (IsAmrish) [reply website/src/pages/Blog/Author/index.tsx]: yes, we are checking the data inside of the useEffect

File: website/src/pages/Blog/Author/index.tsx [hunk 3]
```diff
@@ -45,17 +45,17 @@ const Blog: React.FC = () => {
...
-      id: 0,
-      slug: "" }
-  ]);
+  const [authorBlogsData, setAuthorBlogData] = useState<blog[]>([
+      { title: "",
+        author: "",
+        excerpt: "",
+        author_info: "",
+        date: "",
+        tags: [""],
```
Comment (satyapriyamishra222) [inline website/src/pages/Blog/Author/index.tsx:54]: [] should be good enough I believe.

File: website/src/pages/Blog/Author/index.tsx [hunk 4]
```diff
@@ -64,22 +64,21 @@ const Blog: React.FC = () => {
 
   const fetchBlogs = async () => {
     const { default: blogs } = await import(`../../../posts.json`);
-    setJsonMdData(blogs);
+    const filteredBlogsByAuthorName = blogs.filter(
+      (blog: blog) => blog.author.toLowerCase().replace(/[^\w ]+/g,'').replace(/ +/g,'-') === authorName
```
Comment (satyapriyamishra222) [inline website/src/pages/Blog/Author/index.tsx:68]: Is the variable name & interface name same here ?

File: website/src/pages/Blog/Author/index.tsx [hunk 5]
```diff
@@ -64,22 +64,21 @@ const Blog: React.FC = () => {
...
   };
 
   useEffect(() => {
     fetchBlogs();
-  },[]);
-
-  const filteredAuthorData = (jsonMdData || []).filter(
-    (blog: blog) => blog.author.toLowerCase().replace(/[^\w ]+/g,'').replace(/ +/g,'-') === authorName
-  );
+  },[]); // eslint-disable-line react-hooks/exhaustive-deps
```
Comment (satyapriyamishra222) [inline website/src/pages/Blog/Author/index.tsx:75]: Is there no way to fix this lint issue ?

**Review by satyapriyamishra222 — APPROVED** [review]

**Review by Pallavi-PH — APPROVED** [review]

**Review by rakeshPRaghu — APPROVED** [review]
LGTM

File: website/src/pages/Blog/Author/index.tsx [hunk 6]
```diff
@@ -89,56 +87,56 @@ const Blog: React.FC = () => {
...
-                  alt={filteredAuthorData[0]?.author}
-                  src={`/images/blog/authors/${getAvatar(filteredAuthorData[0]?.author)}.png`}
+                  alt={authorBlogsData[0]?.author}
+                  src={`/images/blog/authors/${getAvatar(authorBlogsData[0]?.author)}.png`}
                   className={classes.large}
                 />
-                <div>
-                  <h1 className={classes.authorText}>{filteredAuthorData[0]?.author || authorName}</h1>
-                  <p className={classes.authorDesc}>{filteredAuthorData[0]?.author_info}</p>
-                </div>
```
Comment (rakeshPRaghu) [inline website/src/pages/Blog/Author/index.tsx:128]: @IsAmrish this section needs to be like this

Comment (IsAmrish) [reply website/src/pages/Blog/Author/index.tsx]: Thankyou!

File: website/src/pages/Blog/Author/styles.ts [hunk 7]
```diff
@@ -93,7 +93,7 @@ const useStyles = makeStyles((theme: Theme) => ({
   },
   cardSize: {
     maxWidth: "480px !important",
-    paddingTop: theme.spacing(4),
+    padding: theme.spacing(2),
```
[the remark this answers is not in the document]
Comment (IsAmrish) [reply website/src/pages/Blog/Author/styles.ts:95]: yes, to have proper space around the card

**Pull request merged**